packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx

```diff
@@ -278,7 +278,7 @@ export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState
     return (
       <ContextualMenu
         className={ css('ms-BaseButton-menuHost') }
-        isBeakVisible={ true }
+        isBeakVisible={ typeof menuProps.isBeakVisible === 'undefined' ? true : menuProps.isBeakVisible }
```

Comment:
Could you just spread the whole object, and provide only the specific overrides after?

E.g.

```
<ContextualMenu
  {...menuProps }
  className={ css('ms-BaseButton-menuhost', menuProps.className) }
  target={ this._buttonElement }
/>

There is no reason to override the default directionalHint or manually pass in items.